src/Dependencies/Updater.php

```diff
@@ -41,63 +27,74 @@ private function updateDependency($dep)
 
         $type = explode('/', $dep)[0];
         $name = explode('/', $dep)[1];
+
+        $result = $this->prepareForUpdate();
+        if ($result->isErr()) {
+            return $result;
+        }
+
         $depArray = $this->jsonFile->getDependency($type, $name);
         if ($depArray === []) {
             return \Result\Result::err('No matching dependency in whippet.json');
         }
 
-        $this->updateHash();
-        $this->loadGitignore();
+        return $this->update([$type=>[$depArray]]);
+    }
 
-        echo sprintf("[Updating %s/%s]\n", $type, $depArray['name']);
-        $result = $this->addDependencyToLockfile($type, $depArray);
+    public function updateAll()
+    {
+        $result = $this->prepareForUpdate();
```

Comment:
Yeah, I did look at getting rid of `prepareForUpdate()` altogether, but both `updateAll()` and `updateSingle()` need to load the Whippet files before they get as far as the generic `update()` method. The only way I could see around that was to load the Whippet files in `Dependencies` and then pass them through to `Updater`, but that felt like a huge refactoring for relatively little benefit.

I have extracted `updateHash()` and `loadGitignore()` to `update()` though.